File: backend/internal/models/inscription.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Inscription struct {
	ID                uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	UserID            uuid.UUID `json:"user_id" gorm:"type:uuid;not null;index"`
	User              User      `json:"-" gorm:"foreignKey:UserID"`
	CongressID        uuid.UUID `json:"congress_id" gorm:"type:uuid;not null;index"`
	Congress          Congress  `json:"-" gorm:"foreignKey:CongressID"`
	Nom               string    `json:"nom" gorm:"not null"`
	Prenom            string    `json:"prenom" gorm:"not null"`
	Email             string    `json:"email" gorm:"not null"`
	Telephone         string    `json:"telephone" gorm:"not null"`
	Organisme         string    `json:"organisme" gorm:"not null"`
	Pays              string    `json:"pays" gorm:"not null"`
	ParticipationType string    `json:"participation_type" gorm:"not null"`
	Montant           float64   `json:"montant" gorm:"not null"`
	MethodePaiement   string    `json:"methode_paiement" gorm:"not null"`

	NumeroFacture     string    `json:"numero_facture" gorm:"uniqueIndex;not null"`
	TransactionID     string    `json:"transaction_id" gorm:""`
	PaymentStatus     string    `json:"payment_status" gorm:"default:pending"`
	CreatedAt         time.Time `json:"created_at"`
	UpdatedAt         time.Time `json:"updated_at"`
}

File: mobile/lib/models/inscription.dart
class Inscription {
  final int id;
  final String userId;
  final String congressId;
  final CongressInfo? congress;
  final String nom;
  final String prenom;
  final String email;
  final String telephone;
  final String? organisme;
  final String pays;
  final String tariffLabel;
  final String participationType;
  final double montant;
  final String methodePaiement;
  final String numeroFacture;
  final String? transactionId;
  final String paymentStatus;
  final DateTime createdAt;

  Inscription({
    required this.id,
    required this.userId,
    required this.congressId,
    this.congress,
    required this.nom,
    required this.prenom,
    required this.email,
    required this.telephone,
    this.organisme,
    required this.pays,
    required this.tariffLabel,
    required this.participationType,
    required this.montant,
    required this.methodePaiement,
    required this.numeroFacture,
    this.transactionId,
    required this.paymentStatus,
    required this.createdAt,
  });

  String get fullName => '$prenom $nom';
  bool get isConfirmed => paymentStatus == 'confirmed';
  bool get isPending => paymentStatus == 'pending';

  factory Inscription.fromJson(Map<String, dynamic> json) {
    CongressInfo? congress;
    if (json['congress'] != null && json['congress'] is Map) {
      congress = CongressInfo.fromJson(json['congress'] as Map<String, dynamic>);
    } else {
      congress = null;
    }

    return Inscription(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id']?.toString() ?? '0') ?? 0,
      userId: json['user_id']?.toString() ?? '',
      congressId: json['congress_id']?.toString() ?? '',
      congress: congress,
      nom: json['nom']?.toString() ?? '',
      prenom: json['prenom']?.toString() ?? '',
      email: json['email']?.toString() ?? '',
      telephone: json['telephone']?.toString() ?? '',
      organisme: json['organisme']?.toString(),
      pays: json['pays']?.toString() ?? '',
      tariffLabel: json['tariff_label']?.toString() ?? '',
      participationType: json['participation_type']?.toString() ?? '',
      montant: (json['montant'] is num ? (json['montant'] as num).toDouble() : double.tryParse(json['montant']?.toString() ?? '0') ?? 0),
      methodePaiement: json['methode_paiement']?.toString() ?? '',
      numeroFacture: json['numero_facture']?.toString() ?? '',
      transactionId: json['transaction_id']?.toString(),
      paymentStatus: json['payment_status']?.toString() ?? 'pending',
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'].toString()) ?? DateTime.now()
          : DateTime.now(),
    );
  }
}

class CongressInfo {
  final String id;
  final String title;
  final String? subtitle;
  final DateTime? startDate;
  final DateTime? endDate;
  final String? location;
  final String? city;
  final String? country;

  CongressInfo({
    required this.id,
    required this.title,
    this.subtitle,
    this.startDate,
    this.endDate,
    this.location,
    this.city,
    this.country,
  });

  String get dateRange {
    if (startDate == null || endDate == null) return '';
    return '${_fmt(startDate!)} - ${_fmt(endDate!)}';
  }

  static String _fmt(DateTime d) =>
      '${d.day.toString().padLeft(2, '0')}/${d.month.toString().padLeft(2, '0')}/${d.year}';

  factory CongressInfo.fromJson(Map<String, dynamic> json) {
    return CongressInfo(
      id: json['id']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      subtitle: json['subtitle']?.toString(),
      startDate: json['start_date'] != null
          ? DateTime.tryParse(json['start_date'].toString())
          : null,
      endDate: json['end_date'] != null
          ? DateTime.tryParse(json['end_date'].toString())
          : null,
      location: json['location']?.toString(),
      city: json['city']?.toString(),
      country: json['country']?.toString(),
    );
  }
}
